### PR TITLE
resolve onclick data type in bitnav (#816)

### LINKED ISSUE #816: Wrong data type for the OnClick property of BitNavLinkItem
The data type of the BitNavLinkItem.OnClick is `EventCallback` which is a blazor-specific data type and should not be used in a plain C# object like here in the `BitNavLinkItem` class.
The correct data type is something like `Action` or `Action<T>`

### PR DESCRIPTION
**Pull request by msynk**
this closes #816

### CODE REVIEW

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor [hunk 1]
```diff
@@ -66,12 +67,12 @@
 
             @if (navLinkItem.Url.HasValue())
             {
-                <a href="@(navLinkItem.Disabled ? "/" : navLinkItem.Url)"
+                <a href="@(navLinkItem.IsEnabled ? navLinkItem.Url : "/")"
+                   style="padding: 0px 20px 0px @(((navLinkItem.Depth * 14) + 27).ToString())px;"
```
Comment (maryamhdr) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor:71]: please remove the `px` unit where the value of the CSS property is equal to 0.

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor [hunk 2]
```diff
@@ -88,8 +89,8 @@
                 if (navLinkItem.ForceAnchor)
                 {
                     <a class="@GetLinkClass(navLinkItem)"
+                       style="padding: 0px 20px 0px @(((navLinkItem.Depth * 14) + 27).ToString())px;"
```
Comment (maryamhdr) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor:92]: please remove the px unit where the value of the CSS property is equal to 0.

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor [hunk 3]
```diff
@@ -100,7 +101,7 @@
                 {
                     <button class="@GetLinkClass(navLinkItem)"
                             title="@(navLinkItem.Title.HasValue() ? navLinkItem.Title : navLinkItem.Name)"
-                            style="padding: 0px 20px 0px @(((navLinkItem.Depth * 14)+27).ToString())px;"
+                            style="padding: 0px 20px 0px @(((navLinkItem.Depth * 14) + 27).ToString())px;"
```
Comment (maryamhdr) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor:104]: please remove the px unit where the value of the CSS property is equal to 0.

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNavLinkItem.cs [hunk 4]
```diff
@@ -58,23 +61,24 @@ public class BitNavLinkItem
...
-        public List<BitNavLinkItem>? Links { get; set; }
+        public IEnumerable<BitNavLinkItem> Links { get; set; } = new List<BitNavLinkItem>();
 
         /// <summary>
         /// Callback invoked when a link in the navigation is clicked
         /// </summary>
-        public EventCallback? OnClick { get; set; }
+        public Action<BitNavLinkItem>? OnClick { get; set; }
+
         internal int Depth { get; set; }
```
Comment (maryamhdr) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNavLinkItem.cs:82]: please add a definition comment for the `Depth` property.

**Review by maryamhdr — CHANGES_REQUESTED** [review]

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor (hunk 1 above)
Comment (msynk) [reply src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor:71]: done

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor (hunk 2 above)
Comment (msynk) [reply src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor:92]: done

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor (hunk 3 above)
Comment (msynk) [reply src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor:104]: done

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNavLinkItem.cs (hunk 4 above)
Comment (msynk) [reply src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNavLinkItem.cs:82]: the access modifier of this property is `internal`, so it is not required for this prop to have documentation comment.

**Review by maryamhdr — APPROVED** [review]

**Pull request merged**